modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/MediaSessionManager.kt

```diff
@@ -484,7 +486,14 @@ class MediaSessionManager(
 
         private fun onAddBookmark() {
             logEvent("add bookmark")
-            bookmarkHelper.handleAddBookmarkAction(context)
+            val coroutineContext = CoroutineScope(Dispatchers.Main + Job())
+            coroutineContext.launch {
+                Util.isAndroidAutoConnectedFlow(context).collect {
+                    bookmarkHelper.handleAddBookmarkAction(context, it)
+                    // Cancel the coroutine after the bookmark has been added
+                    coroutineContext.cancel()
+                }
```

Comment:
I wonder if there's any risk that `Util.isAndroidAutoConnectedFlow(context)` never emits an item, resulting in the bookmark never getting added. That shouldn't happen, but we are dependent on that flow working correctly. I wonder if it might be worth adding a `Util.isAndroidAutoConnected` method to the Util class that just directly returns a nullable value so that there is no risk of getting stuck here:

``` kotlin
fun isAndroidAutoConnected(context: Context) =
    CarConnection(context).type.value
```

I don't feel strongly. What we have here should be fine. Just a thought.

Reply:
` CarConnection(context).type` is a `LiveData`, I think we still need to observe it to get the value. Let me ponder over it and discuss it with you. I'll go ahead and merge this one and if needed, I can raise another pull request.